daemon/logger/jsonfilelog/read.go

```diff
@@ -256,7 +256,7 @@ func followLogs(f *os.File, logWatcher *logger.LogWatcher, notifyRotate chan int
 
 	handleDecodeErr := func(err error) error {
 		if err == io.EOF {
-			for err := waitRead(); err != nil; {
+			for err := waitRead(); err != nil; err = waitRead() {
```

Comment:
this construction looks extremely confusing for me, why don't just use for {} and do stuff inside?